gpbft/gpbft.go

```diff
@@ -528,9 +548,18 @@ func (i *instance) beginCommit() {
 	// No justification is required for committing bottom.
 	var justification *Justification
 	if !i.value.IsZero() {
-		if quorum, ok := i.getRound(i.current.Round).prepared.FindStrongQuorumFor(i.value.Key()); ok {
+		valueKey := i.value.Key()
+		currentRound := i.getRound(i.current.Round)
+		nextRound := i.getRound(i.current.Round + 1)
+		if quorum, ok := currentRound.prepared.FindStrongQuorumFor(valueKey); ok {
 			// Found a strong quorum of PREPARE, build the justification for it.
 			justification = i.buildJustification(quorum, i.current.Round, PREPARE_PHASE, i.value)
+		} else if justifiedByCommit := currentRound.committed.GetJustificationOf(PREPARE_PHASE, valueKey); justifiedByCommit != nil {
```

Comment:
We are refetching justifications here which is aligned with the current pattern of method call in core gpbft package. 

I ideally we should pass in a justification to `beginCommit` to avoid looking it up twice. (similar story for prepare).

But for now I have kept the diff small in favour of getting the logic correct.